### PR TITLE
[ADVAPP-345]: Track views of knowledge base articles

### PR DESCRIPTION
**Pull request by danharrin**
### Ticket(s) or GitHub Issue

- https://canyongbs.atlassian.net/browse/ADVAPP-345

### Technical Description

Rate limiter applied to the "view" page of the View Knowledge Base Item page for 1 attempt every minute. It only gets bumped when the page gets loaded and not when any Livewire update request gets made.

### CODE REVIEW

File: app-modules/knowledge-base/database/migrations/2024_03_04_155253_create_knowledge_base_item_views_table.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+return new class () extends Migration {
+    public function up(): void
+    {
+        Schema::create('knowledge_base_item_views', function (Blueprint $table) {
+            $table->uuid('id')->primary();
+            $table->foreignUuid('knowledge_base_item_id')->constrained('knowledge_base_articles')->cascadeOnDelete();
+            $table->foreignUuid('user_id')->constrained()->cascadeOnDelete();
```
Comment (Orrison) [inline app-modules/knowledge-base/database/migrations/2024_03_04_155253_create_knowledge_base_item_views_table.php:47]: Let's actually remove the `cascadeOnDelete` for the User here. I feel like even if the User record was deleted we wouldn't want to decrement views.

File: app-modules/knowledge-base/src/Filament/Resources/KnowledgeBaseItemResource/Pages/ViewKnowledgeBaseItem.php [hunk 2]
```diff
@@ -56,6 +58,19 @@ public function getTitle(): string | Htmlable
         return $this->record->title;
     }
 
+    public function mount(int | string $record): void
```
Comment (Orrison) [inline app-modules/knowledge-base/src/Filament/Resources/KnowledgeBaseItemResource/Pages/ViewKnowledgeBaseItem.php:61]: I think we should maybe do this on the edit page as well. What do you think?

**Review by Orrison — CHANGES_REQUESTED** [review]

File: app-modules/knowledge-base/database/migrations/2024_03_04_155253_create_knowledge_base_item_views_table.php (hunk 1 above)
Comment (danharrin) [reply app-modules/knowledge-base/database/migrations/2024_03_04_155253_create_knowledge_base_item_views_table.php:47]: So it needs to be nullable and nullOnDelete?

File: app-modules/knowledge-base/src/Filament/Resources/KnowledgeBaseItemResource/Pages/ViewKnowledgeBaseItem.php (hunk 2 above)
Comment (danharrin) [reply app-modules/knowledge-base/src/Filament/Resources/KnowledgeBaseItemResource/Pages/ViewKnowledgeBaseItem.php:61]: That wasn't part of the ticket, and it feels like editors aren't utilising the article so it shouldn't increment the views IMO

File: app-modules/knowledge-base/database/migrations/2024_03_04_155253_create_knowledge_base_item_views_table.php (hunk 1 above)
Comment (danharrin) [reply app-modules/knowledge-base/database/migrations/2024_03_04_155253_create_knowledge_base_item_views_table.php:47]: Done that, but LMK if you want it some other way

File: app-modules/knowledge-base/src/Filament/Resources/KnowledgeBaseItemResource/Pages/ViewKnowledgeBaseItem.php (hunk 2 above)
Comment (Orrison) [reply app-modules/knowledge-base/src/Filament/Resources/KnowledgeBaseItemResource/Pages/ViewKnowledgeBaseItem.php:61]: Got it, makes sense.

**Review by Orrison — APPROVED** [review]

**Pull request merged**